### PR TITLE
docs: design — Diplomatic Relations System (Wave 1)

### PR DESCRIPTION
**Pull request by ianlintner**
## Summary

Adds the design doc for a new **Diplomatic Relations System** that extends the existing dilemma pattern with player-initiated diplomacy actions (gift, lobby, propose non-compete, surveil) plus throttled AI-initiated offers. Wave 1 ships a symmetric thin slice across empires and rival companies; sabotage, contract poaching, and tech-improved espionage are explicitly deferred to wave 2.

Spec: [\`docs/plans/2026-04-30-diplomatic-relations-design.md\`](docs/plans/2026-04-30-diplomatic-relations-design.md)

## Key design decisions

- **Hub + throttled offers** — A new \`DiplomacyScene\` (Foreign Relations hub) is the player's home for initiating actions, while AI offers continue to surface through the existing \`pendingChoiceEvents[]\` queue (≤1/turn, sharing the dilemma slot).
- **Standing + tags model** — Per target: a single \`0..100\` standing number (temperature) plus durable narrative tags (\`OweFavor\`, \`RecentlyGifted\`, \`SuspectedSpy\`, \`NonCompete\`, \`LeakedIntel\`). Reuses the existing-but-unwired \`empireReputation\` field, promoted into \`diplomacy.empireStanding\`.
- **Ambassadors recur, rulers are rare** — Each empire/rival gets a generated ambassador or relations director for routine dialog. Ruler/CEO portraits surface only for tier transitions, espionage exposure, and formal offers — preserving their gravitas.
- **Hard spam cap** — Per turn, total diplomacy modal volume ≤ 3 (≤2 player-initiated outcomes that breach the modal threshold + ≤1 AI offer). Routine outcomes route to a new "Diplomatic activity" digest section in the review screen. Tier transitions, espionage exposure, and non-compete acceptance/refusal always earn a modal.
- **Wave 1 verbs** — Send Gift (empire/rival), Lobby For/Against (empire targeting a rival), Propose Non-Compete (rival targeting an empire pair), Surveil (rival, single payload: cash / top contract by value / top empire-standing for 3 turns).

## What's not in this PR

This is **design-only** — no code yet. Implementation plan and code land in follow-up PRs once the design is reviewed.

## Reviewer notes

- The five sharpening questions (engagement model, MVP scope, standing shape, character layer, cadence/cost, espionage payload) and their tradeoffs are documented in §1–§6 of the spec.
- Save compatibility: \`empireReputation\` is currently stubbed but unwired in gameplay, so promotion to \`diplomacy.empireStanding\` is safe in this branch. A one-shot migration in \`loadGame\` is described in §11 if any wild save data exists.
- Open questions surfaced in §13 with documented defaults (auto-spent \`OweFavor\` vs. explicit "call in favor" verb; small badge vs. summary card on the planning HUD).

## Test plan

- [ ] Read the design doc end-to-end and flag any unclear sections, missing edge cases, or balance concerns
- [ ] Confirm the modal-vs-digest split feels right for "tight loop, no spam"
- [ ] Confirm the wave 1 / wave 2 cut matches your intent
- [ ] Confirm the ambassador layer (vs. ruler-only) is worth the additional character generation surface

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by ianlintner** [comment]
## RESULT — review concerns addressed

All 7 actionable concerns from the formal review at [issuecomment-4352609920](https://github.com/ianlintner/space-tycoon/pull/253#issuecomment-4352609920) are addressed in commit [`5242914`](https://github.com/ianlintner/space-tycoon/pull/253/commits/5242914) (with merge of `main` in `e4a35e2` resolving the prior conflicts).

| # | Concern | Resolution |
|---|---|---|
| 1 | `decrementCooldowns` misnamed | Renamed to `pruneExpiredCooldowns`; updated callers + tests |
| 2 | `addTag` allows duplicates | Added `replaceTag`; switched `RecentlyGifted` / `OweFavor` to use it |
| 3 | Cap not enforced | Cap is enforced via `slice(0, cap)` in `processQueuedDiplomacyActions`; contract documented in JSDoc; `tickDiplomacyState` resets `actionsResolvedThisTurn` |
| 4 | Hardcoded `"human"` portraits | Inherits `PortraitCategory` from empire leader / rival CEO; new test asserts inheritance |
| 5 | Raw IDs in digest text | ID→display-name substitution at the dispatcher boundary |
| 6 | Shared `[]` reference in `EMPTY_DIPLOMACY_STATE` | `Object.freeze` applied to the constant and its `queuedActions` array |
| 7 | Missing diplomacy seed test | Already present in `src/game/__tests__/NewGameSetup.test.ts:169-186` from the original Task 7 |
| 8 | Undocumented `OweFavor` asymmetry | Comment added in `resolveGiftRival` explaining wave-1 design intent |

CI gates (post-fix): `npm run typecheck` clean, `npm run test` 1377/1377 passing, `npm run build` succeeds. The 3 caretaker task pings at 16:27Z all reference this same pre-fix review state and the fix at 16:24Z preceded them by 3 minutes — no new work is required.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: src/game/diplomacy/Cooldowns.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export function setCooldown(
+  cooldowns: Record<string, number>,
+  key: string,
+  nextAvailableTurn: number,
+): Record<string, number> {
+  return { ...cooldowns, [key]: nextAvailableTurn };
+}
+
+export function decrementCooldowns(
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/game/diplomacy/Cooldowns.ts]: Addressed in 5242914 — renamed to `pruneExpiredCooldowns` (Cooldowns.ts:28); callers in DiplomacyTick.ts and the test file updated.

File: src/game/diplomacy/StandingTags.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import type { StandingTag } from "../../data/types.ts";
+
+export function addTag(
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/game/diplomacy/StandingTags.ts:3]: Addressed in 5242914 — added `replaceTag` (StandingTags.ts:14) and switched `RecentlyGifted` and `OweFavor` in resolveGiftEmpire/resolveGiftRival to use it.

File: src/game/diplomacy/DiplomacyResolver.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    nextState: {
+      ...state,
+      cash: cashAfter,
+      empireReputation: {
+        ...(state.empireReputation ?? {}),
+        [targetId]: nextStanding,
+      },
+      diplomacy: {
+        ...prevDip,
+        empireTags: {
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/game/diplomacy/DiplomacyResolver.ts:151]: Addressed in 5242914 — cap is enforced via `slice(0, cap)` in `processQueuedDiplomacyActions`; contract documented in JSDoc; `tickDiplomacyState` resets `actionsResolvedThisTurn` each turn.

File: src/game/diplomacy/AmbassadorGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  "amb-07",
+  "amb-08",
+];
+
+function makeAmbassador(rng: SeededRNG): Ambassador {
+  const first = rng.pick([...FIRST_NAMES]);
+  const last = rng.pick([...SURNAMES]);
+  return {
+    name: `${first} ${last}`,
+    portrait: { portraitId: rng.pick([...PORTRAIT_IDS]), category: "human" },
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/game/diplomacy/AmbassadorGenerator.ts]: Addressed in 5242914 — `generateAmbassadors` now inherits `PortraitCategory` from `empire.leaderPortrait.category` and `rival.ceoPortrait.category`. New test asserts category inheritance.

File: src/game/diplomacy/DiplomacyResolver.ts [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+  if (success) {
+    const delta = Math.floor(GIFT_EMPIRE_BASE_DELTA * dampener);
+    nextStanding = applyStandingDelta(beforeStanding, delta);
+    tagsAfter = addTag(tagsAfter, {
+      kind: "RecentlyGifted",
+      expiresOnTurn: state.turn + GIFT_RECENTLY_GIFTED_TTL,
+    });
+    if (rng.chance(GIFT_OWE_FAVOR_CHANCE)) {
+      tagsAfter = addTag(tagsAfter, {
+        kind: "OweFavor",
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/game/diplomacy/DiplomacyResolver.ts:109]: Addressed in 5242914 — `processQueuedDiplomacyActions` builds a name lookup from `state.galaxy.empires` and `state.aiCompanies`, then substitutes IDs in digest text and modal flavor/headline.

File: src/data/types.ts [hunk 6]
```diff
@@ -1071,6 +1072,86 @@ export interface CharacterPortrait {
...
+  empireTags: Record<string, readonly StandingTag[]>;
+  rivalTags: Record<string, readonly StandingTag[]>;
+  empireAmbassadors: Record<string, Ambassador>;
+  rivalLiaisons: Record<string, Ambassador>;
+  cooldowns: Record<string, number>;
+  queuedActions: readonly QueuedDiplomacyAction[];
+  actionsResolvedThisTurn: number;
+}
+
+export const EMPTY_DIPLOMACY_STATE: DiplomacyState = {
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/data/types.ts]: Addressed in 5242914 — `Object.freeze` applied to both `EMPTY_DIPLOMACY_STATE` and its `queuedActions` array (types.ts:1143-1153).

File: src/game/__tests__/NewGameSetup.test.ts [hunk 7]
```diff
@@ -165,3 +165,23 @@ describe("NewGameSetup", () => {
     expect(storyteller.consecutiveProfitTurns).toBe(0);
   });
 });
```
[the remark this answers is not in the document]
Comment (ianlintner) [reply src/game/__tests__/NewGameSetup.test.ts:167]: Already covered — `src/game/__tests__/NewGameSetup.test.ts:169-186` (added in Task 7) asserts `state.diplomacy.empireAmbassadors[empireId]` and `state.diplomacy.rivalStanding[rivalId] === 50` for a generated game.
